src/main/scala/org/broadinstitute/dsde/workbench/sam/api/UserInfoDirectives.scala

```diff
@@ -22,11 +22,30 @@ trait UserInfoDirectives {
   val directoryDAO: DirectoryDAO
   val registrationDAO: RegistrationDAO
   val cloudExtensions: CloudExtensions
+  val tosService: TosService
   val termsOfServiceConfig: TermsOfServiceConfig
 
+  /**
+    * Extracts authentication information from headers, looks up user in database,
+    * returns user only if the user is enabled and has accepted latest terms of service.
+    * Throws 401 exception if user has not accepted latest terms of service or is disabled.
+    * Throws 403 exception if user does not exist (not 404 because that would mean the requested URL does not exist).
+    * @param samRequestContext
+    * @return
+    */
   def requireActiveUser(samRequestContext: SamRequestContext): Directive1[SamUser]
 
-  def requireCreateUser(samRequestContext: SamRequestContext): Directive1[SamUser]
+  /**
+    * Extracts authentication information from headers, looks up user in database,
+    * returns user regardless of enabled or terms of service status.
+    * Specifically named to be clear that inactive users are permitted.
+    * Throws 403 exception if user does not exist (not 404 because that would mean the requested URL does not exist).
+    * @param samRequestContext
+    * @return
+    */
+  def requireUserAllowInactive(samRequestContext: SamRequestContext): Directive1[SamUser]
+
+  def withNewUser(samRequestContext: SamRequestContext): Directive1[SamUser]
 
   def asWorkbenchAdmin(samUser: SamUser): Directive0 =
```

Comment:
See above "renaming directives" comment.  This one would be included too.  We have 3 different prefixes on the Directives, `with`, `require`, `as`.  It doesn't really matter that much, but I feel like it would make this whole trait slightly more easy to understand overall if they followed a common naming convention.